src/controller/java/templates/CHIPClusters-JNI.zapt

```diff
@@ -672,6 +672,65 @@ JNI_METHOD(void, {{asCamelCased ../name false}}Cluster, write{{asCamelCased name
     }
 }
 {{/if}}
+{{#if isReportableAttribute}}
+
+JNI_METHOD(void, {{asCamelCased ../name false}}Cluster, configure{{asCamelCased name false}}Attribute)(JNIEnv * env, jobject self, jlong clusterPtr, jobject callback, jint minInterval, jint maxInterval{{#if isAnalog}}, {{asJniBasicType type}} change{{/if}})
+{
+    StackLockGuard lock(JniReferences::GetInstance().GetStackLock());
+    CHIPDefaultSuccessCallback * onSuccess = new CHIPDefaultSuccessCallback(callback);
```

Comment:
could using std::unique_ptr be easier in managing the lifetime of these callbacks?
it seems as if it  would only need special handling on the success case (call `release` when the ownership was passed along)

what is the lifetime for onSuccess/onFailure? I see them allocated with `new` on the heap here. What will delete them once you call ConfigureAttribute? Generally I expect the SDK to use Platform::New/Delete so wondering here what will ensure that the `delete` operator will be called rather than Platform::Delete (which may be different)?

Reply:
I switched to std::unique_ptr and Platform::New/Delete for all the methods here.

For callback lifetime, there are some issues (#8773) that need to be fixed, but probably not as part of this PR.